### ISSUE TITLE
Add favorite experiments and metrics

### ISSUE THREAD

**Issue by aaronyan** [Automattic/abacus#705]
I think it would be awesome to be able to favorite experiments so they appear at the top of the Experiments Directory. This could also be used for experiment templates, where folks can clone experiments that they commonly use so they don't have to remake similar experiments.

Favorited metrics would appear as the first suggestions when adding new metrics. Teams could suggest to new experimenters which metrics to favorite to reduce the friction during experiment creation.

We would need to brainstorm the technical requirements for these features (where to store the favorites, etc) and estimate the size of the feature.

**Comment by jessie-ross** [comment]
Yeah, a big part of this is that the frontend needs some kind of simple persistent storage. Something like Firebase would be awesome, maybe A8C has a simple KV store somewhere?

**Comment by razvanpapadopol** [comment]
I'm pretty sure favorited metrics would be super helpful. We can even add the possibility to "star" a metric (or even all metrics) directly from the metrics table of an existing experiment! Sometimes you don't want to clone the full experiment but just some of its metrics.

> maybe A8C has a simple KV store somewhere

While I'm not aware of anything like this, as an example, for a WordPress.com user, we're [storing Calypso preferences](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/preferences/actions.js) using WP.COM API `v1.1/me/preferences/` endpoint. 

On staging there's a quick way to view/update them:

<img width="654" alt="Screenshot 2022-05-18 at 21 10 34" src="https://user-images.githubusercontent.com/14192054/169117451-fab3222f-8d16-454f-8724-d610aaa1ce31.png">
